### DISCUSSION TITLE
allow to define AWS_ROLE_ARN for assume-role-with-web-identity

### DISCUSSION THREAD

**Discussion by mo-saeed** [grafana/grafana#66913] [Data Sources]
Hello,

I am trying to implement https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html so datasources in grafana can access AWS elasticsearch.
 
so in grafana pod level, i defined:
```
AWS_ROLE_ARN
AWS_WEB_IDENTITY_TOKEN_FILE
AWS_REGION
```
which works fine.

the problem we have is that we have some datasources need to access ES in different AWS accounts so for each data source i need to define different AWS_ROLE_ARN. I tried to set `assumeRoleARN` in the data source but it didn't work.

is there a way to set `AWS_ROLE_ARN` for assume-role-with-web-identity at the data source level ?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). This is phrased as a question, and we'd like questions about grafana to go to our[ community forum](https://community.grafana.com/). Feel free to make an [issue](https://github.com/grafana/grafana/issues) if you cannot find a solution and would like to request the feature.
